include/oneapi/mkl/blas.hxx

```diff
@@ -261,7 +261,7 @@ static inline void gemm(cl::sycl::queue &queue, transpose transa, transpose tran
                  c, ldc);
     gemm_postcondition(queue, transa, transb, m, n, k, alpha, a, lda, b, ldb, beta, c, ldc);
 }
-
+#ifdef ENABLE_HALF_ROUTINES
```

Comment:
Do we even need to ifdef away this interface entirely? Could we just keep the interface and throw an exception if we detect that we want to run a half function on a device that does not support half (which currently is always in hipSYCL)? I think hipSYCL defines the `half` type to something like a `short`, so the interface by itself should compile.